File: app/src/main/java/com/yechaoa/yutils/SpUtil.java
package com.yechaoa.yutils;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by yechao on 2017/9/11.
 * Describe : SpUtil
 * <p>
 * GitHub : https://github.com/yechaoa
 * CSDN : http://blog.csdn.net/yechaoa
 */
public class SpUtil {

    private static final String FILE_NAME = "config";
    private static SharedPreferences sp = YUtils.getApplication().getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE);

    /**
     * String
     */
    public static void setString(String key, String value) {
        sp.edit().putString(key, value).apply();
    }

    public static String getString(String key) {
        return sp.getString(key, "");
    }

    /**
     * Int
     */
    public static void setInt(String key, int value) {
        sp.edit().putInt(key, value).apply();
    }

    public static int getInt(String key) {
        return sp.getInt(key, 0);
    }

    /**
     * Boolean
     */
    public static void setBoolean(String key, boolean value) {
        sp.edit().putBoolean(key, value).apply();
    }

    public static boolean getBoolean(String key) {
        return sp.getBoolean(key, false);
    }

    /**
     * Float
     */
    public static void setFloat(String key, Float value) {
        sp.edit().putFloat(key, value).apply();
    }

    public static Float getFloat(String key) {
        return sp.getFloat(key, 0);
    }

    /**
     * Long
     */
    public static void setLong(String key, Long value) {
        sp.edit().putLong(key, value).apply();
    }

    public static Long getLong(String key) {
        return sp.getLong(key, 0);
    }

    /**
     * Remove
     */
    public static void removeByKey(String key) {
        sp.edit().remove(key).apply();
    }

    public static void removeAll() {
        sp.edit().clear().apply();
    }

}

File: yutilskt/src/main/java/com/yechaoa/yutilskt/SpUtil.kt
package com.yechaoa.yutilskt

import android.content.Context
import android.content.SharedPreferences


/**
 * Created by yechao on 2020/1/7.
 * Describe : SpUtilKt
 *
 * GitHub : https://github.com/yechaoa
 * CSDN : http://blog.csdn.net/yechaoa
 */
object SpUtil {
    private const val FILE_NAME = "config"
    private val sp: SharedPreferences = YUtils.getApp().getSharedPreferences(FILE_NAME, Context.MODE_PRIVATE)

    /**
     * String
     */
    fun setString(key: String?, value: String?) {
        sp.edit().putString(key, value).apply()
    }

    fun getString(key: String?, defValue: String = ""): String {
        return sp.getString(key, defValue)
    }

    /**
     * StringSet
     */
    fun setStringSet(key: String?, value: Set<String>?) {
        sp.edit().putStringSet(key, value).apply()
    }

    fun getStringSet(key: String?): Set<String> {
        return HashSet<String>(sp.getStringSet(key, HashSet<String>()))
    }

    /**
     * Int
     */
    fun setInt(key: String?, value: Int) {
        sp.edit().putInt(key, value).apply()
    }

    fun getInt(key: String?, defValue: Int = 0): Int {
        return sp.getInt(key, defValue)
    }

    /**
     * Boolean
     */
    fun setBoolean(key: String?, value: Boolean) {
        sp.edit().putBoolean(key, value).apply()
    }

    fun getBoolean(key: String?, defValue: Boolean = false): Boolean {
        return sp.getBoolean(key, defValue)
    }

    /**
     * Float
     */
    fun setFloat(key: String?, value: Float?) {
        sp.edit().putFloat(key, value!!).apply()
    }

    fun getFloat(key: String?, defValue: Float = 0f): Float {
        return sp.getFloat(key, defValue)
    }

    /**
     * Long
     */
    fun setLong(key: String?, value: Long?) {
        sp.edit().putLong(key, value!!).apply()
    }

    fun getLong(key: String?, defValue: Long = 0): Long {
        return sp.getLong(key, defValue)
    }

    /**
     * Remove
     */
    fun removeByKey(key: String?) {
        sp.edit().remove(key).apply()
    }

    fun removeAll() {
        sp.edit().clear().apply()
    }
}